packages/system/src/Name/Components/NameSearchModal/NameSearchModal.tsx

```diff
@@ -7,7 +7,7 @@ interface NameSearchProps {
   open: boolean;
   onClose: () => void;
   onChange: (name: Name) => void;
-
+  onlyShowStores?: boolean;
```

Comment:
there's no filter in the API to do this, so we need to do it in JS. Not such a big deal. Though, this boolean annoys me.

An `extraFilter` prop where we can pass a more generic filtering function was another thought, though getting a list of 'Internal suppliers' is relatively common.

I had a thought that there should just be more components. `CustomerSearchModal`, `SupplierSearchModal`, `InternalSupplierSearchModal` etc instead

Reply:
agree - creating more components would help here.